lib/src/domain/user_cases/implementation/get_top_rated_movies_use_case.dart

```diff
@@ -0,0 +1,19 @@
+import '../../../core/utils/state.dart';
+
+import '../../../data/datasource/remote/i_api_service.dart';
+import '../../../data/datasource/remote/movies_api_provider.dart';
+import '../../../data/repositories/movies_repository.dart';
+
+class GetTopRatedMoviesUseCase {
+  final ApiService api = ApiMovieService();
+  late final MoviesRepository movieRepo = MoviesRepository(
+    api: api,
+  );
```

Comment:
Same as above

Reply:
If you notice you are replicating the same functionality 4 times, using the repository and send it as a parameter the enpoint.
You have two options here, create a unique UseCase that as a parameter has the url like
```
GetMoviesByEndpointUseCase({
     required this.repository,
     required this.endpoint,
});

final IRepository repository;
final String endpoint;

Future<DataState> getData() async {
    return await movieRepo.getData(endpoint);
  }
```

Reply:
You are right! Nice!